src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -146,69 +138,74 @@ class NexusPublishPlugin : Plugin<Project> {
-            description = "Gets and displays a staging profile id for a given repository and package group. " +
+            it.group = PublishingPlugin.PUBLISH_TASK_GROUP
+            it.description = "Gets and displays a staging profile id for a given repository and package group. " +
                 "This is a diagnostic task to get the value and " +
                 "put it into the NexusRepository configuration closure as stagingProfileId."
-            repository.convention(repo)
-            packageGroup.convention(extension.packageGroup)
+            it.repository.convention(repo)
+            it.packageGroup.convention(extension.packageGroup)
         }
-        val initializeTask = tasks.register<InitializeNexusStagingRepository>(
-            "initialize${repo.capitalizedName}StagingRepository"
+        val initializeTask = tasks.register(
+            "initialize${repo.capitalizedName}StagingRepository",
+            InitializeNexusStagingRepository::class.java
         ) {
-            group = PublishingPlugin.PUBLISH_TASK_GROUP
-            description = "Initializes the staging repository in '${repo.name}' Nexus instance."
-            registry.set(registryService)
-            usesService(registryService)
-            repository.convention(repo)
-            packageGroup.convention(extension.packageGroup)
+            it.group = PublishingPlugin.PUBLISH_TASK_GROUP
+            it.description = "Initializes the staging repository in '${repo.name}' Nexus instance."
+            it.registry.set(registryService)
+            it.usesService(registryService)
+            it.repository.convention(repo)
+            it.packageGroup.convention(extension.packageGroup)
         }
-        val findStagingRepository = tasks.register<FindStagingRepository>(
-            "find${repo.capitalizedName}StagingRepository"
+        val findStagingRepository = tasks.register(
+            "find${repo.capitalizedName}StagingRepository",
+            FindStagingRepository::class.java
         ) {
-            group = PublishingPlugin.PUBLISH_TASK_GROUP
-            description = "Finds the staging repository for ${repo.name}"
-            registry.set(registryService)
-            usesService(registryService)
-            repository.convention(repo)
-            packageGroup.convention(extension.packageGroup)
-            descriptionRegex.convention(extension.repositoryDescription.map { "\\b" + Regex.escape(it) + "(\\s|$)" })
+            it.group = PublishingPlugin.PUBLISH_TASK_GROUP
+            it.description = "Finds the staging repository for ${repo.name}"
+            it.registry.set(registryService)
+            it.usesService(registryService)
+            it.repository.convention(repo)
+            it.packageGroup.convention(extension.packageGroup)
+            it.descriptionRegex.convention(extension.repositoryDescription.map { repoDescription -> "\\b" + Regex.escape(repoDescription) + "(\\s|$)" })
```

Comment:
`it`'s the reason why I didn't remove it when trying Gradle 8.3

Reply:
Can do `with(it) {}` if the extra nesting doesn't bother anyone

Reply:
I would stick with `it.` then.

Reply:
Sidenote: we could also apply the SAM compiler plugin manually to get this feature back (I played around in the past, most of the things kotlin-dsl does is possible without it.)

Reply:
> Sidenote: we could also apply the SAM compiler plugin manually to get this feature back (I played around in the past, most of the things kotlin-dsl does is possible without it.)

Are the any side effects/downsides?

Reply:
> Are the any side effects/downsides?

Not sure, I think I abandoned it in the end, because I did something else.
As far as I know it compiles down to the right code in the end, so the only downside I see is that it complicates the build setup of this project, but shouldn't affect consumers.

Probably the best to just go with this PR and then reintroduce pieces separately if we can.